usaspending_api/recipient/delta_models/recipient_profile.py

```diff
@@ -1,26 +1,436 @@
+            WHEN COALESCE(fpds.awardee_or_recipient_uniqu, fabs.awardee_or_recipient_uniqu) IS NOT NULL THEN CONCAT('duns-', COALESCE(fpds.awardee_or_recipient_uniqu, fabs.awardee_or_recipient_uniqu))
+            ELSE CONCAT('name-', COALESCE(fpds.awardee_or_recipient_legal, fabs.awardee_or_recipient_legal, '')) END
+        )) AS recipient_hash,
+        COALESCE(fpds.awardee_or_recipient_uniqu, fabs.awardee_or_recipient_uniqu) AS recipient_unique_id,
+        COALESCE(fpds.ultimate_parent_unique_ide, fabs.ultimate_parent_unique_ide) AS parent_recipient_unique_id,
+        COALESCE(fpds.awardee_or_recipient_uei, fabs.uei) AS uei,
+        COALESCE(fpds.ultimate_parent_uei, fabs.ultimate_parent_uei) AS parent_uei,
+        CASE
+            WHEN tn.type IN ('A', 'B', 'C', 'D')      THEN 'contract'
+            WHEN tn.type IN ('02', '03', '04', '05')  THEN 'grant'
+            WHEN tn.type IN ('06', '10')              THEN 'direct payment'
+            WHEN tn.type IN ('07', '08')              THEN 'loans'
+            WHEN tn.type IN ('09', '11')              THEN 'other'     -- collapsing insurance into other
+            WHEN tn.type LIKE 'IDV%'                  THEN 'contract'  -- collapsing idv into contract
+            ELSE NULL
+        END AS award_category,
+        CASE
+            WHEN COALESCE(fpds.ultimate_parent_uei, fabs.ultimate_parent_uei) IS NOT NULL THEN 'C'
+            ELSE 'R' END AS recipient_level,
+        tn.action_date,
+        CAST(COALESCE(CASE
+            WHEN tn.type IN('07','08') THEN tn.original_loan_subsidy_cost
+            ELSE tn.federal_action_obligation
+        END, 0) AS NUMERIC(23, 2)) AS generated_pragmatic_obligation
+    FROM
+        raw.transaction_normalized as tn
+    LEFT OUTER JOIN raw.transaction_fpds as fpds ON tn.id = fpds.transaction_id
+    LEFT OUTER JOIN raw.transaction_fabs as fabs ON tn.id = fabs.transaction_id
+    WHERE
+        tn.action_date >= '2007-10-01'
+        AND tn.type IS NOT NULL
+    );""",
+    #
+    # --------------------------------------------------------------------------------
+    # -- Step 2, Create the new table and populate with 100% of combinations
+    # --------------------------------------------------------------------------------
+    """CREATE OR REPLACE TABLE temporary_restock_recipient_profile (
+        id BIGINT,
+        recipient_level character(1) NOT NULL,
+        recipient_hash STRING,
+        recipient_unique_id STRING,
+        uei STRING,
+        parent_uei STRING,
+        recipient_name STRING,
+        unused BOOLEAN,
+        recipient_affiliations ARRAY<STRING>,
+        award_types ARRAY<STRING>,
+        last_12_months NUMERIC(23,2),
+        last_12_contracts NUMERIC(23,2),
+        last_12_grants NUMERIC(23,2),
+        last_12_direct_payments NUMERIC(23,2),
+        last_12_loans NUMERIC(23,2),
+        last_12_other NUMERIC(23,2),
+        last_12_months_count INT
+    ) USING DELTA;
```

Comment:
Does a location need to be specified for the temporary table? Also probably good to specify which DB the table will be created in.